docker-style/action.yml

```diff
@@ -0,0 +1,133 @@
+
+  .. note::
+
+      This action emphasizes the fact of having Dockerfile(s) contained inside the
+      'docker' directory in the root of the project.
+
+  .. warning::
+
+      This action only looks for docker files named Dockerfile.
+      A docker file like Dockerfile.linux will not be linted.
+
+inputs:
+
+  # Optional inputs
+
+  directory:
+    description: >
+      Directory from which to search for Dockerfile(s).
+    required: false
+    default: docker
+    type: string
+
+  recursive:
+    description: >
+      Search for Dockerfile(s) recursively.
+    required: false
+    default: false
+    type: bool
+
+  error-level:
+    description: >
+      Fail action based on hadolint output (-1: never, 0: error, 1: warning,
+      2: info)
+    required: false
+    default: 2
+    type: int
+
+runs:
+  using: "composite"
+  steps:
+
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Check that directory 'docker' exists at the root of the repository.
+          If it does not, then a warning is emitted.
+
+    - name: "Check if docker directory exists"
+      shell: bash
+      run: |
+        if [ -d "${{ github.workspace }}/docker" ]; then
+          echo "HAS_DOCKER_DIR=true" >> $GITHUB_ENV
+        else
+          echo "HAS_DOCKER_DIR=false" >> $GITHUB_ENV
+        fi
+
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      if: env.HAS_DOCKER_DIR == 'false'
+      with:
+        level: "WARNING"
+        message: >
+          No 'docker' directory found. Except in special cases, e.g.
+          '.devcontainer', we emphasize that docker files are stored in the
+          'docker' directory at the root of the repository.
+
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Check that directory
+          '${{ github.workspace }}/${{ inputs.directory }}' exists. If not,
+          a warning is raised and nothing is done.
+
+    - name: "Check if directory input exists"
+      shell: bash
+      run: |
+        if [ -d "${{ github.workspace }}/${{ inputs.directory }}" ]; then
+          echo "DIRECTORY_EXISTS=true" >> $GITHUB_ENV
+        else
+          echo "DIRECTORY_EXISTS=false" >> $GITHUB_ENV
+        fi
+
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      if: env.DIRECTORY_EXISTS == 'true'
+      with:
+        level: "INFO"
+        message: >
+          Lint directory '${{ github.workspace }}/${{ inputs.directory }}'.
+
+    - name: "Run Hadolint"
+      if: env.DIRECTORY_EXISTS == 'true'
+      uses: jbergstroem/hadolint-gh-action@v1
+      with:
+        dockerfile: >-
```

Comment:
Interesting... why the dash? ``>-``

Reply:
It was to use "Folded Style" to create a "long" string in a more readable way by spreading it over several lines.
It might be possible to do it in another way but I didn't know it!